eth/catalyst/api.go

```diff
@@ -184,47 +184,37 @@ func (api *ConsensusAPI) ForkchoiceUpdatedV1(update engine.ForkchoiceStateV1, pa
 }
 
 // ForkchoiceUpdatedV2 is equivalent to V1 with the addition of withdrawals in the payload attributes.
-func (api *ConsensusAPI) ForkchoiceUpdatedV2(update engine.ForkchoiceStateV1, payloadAttributes *engine.PayloadAttributes) (engine.ForkChoiceResponse, error) {
-	if payloadAttributes != nil {
-		if err := api.verifyPayloadAttributes(payloadAttributes); err != nil {
-			return engine.STATUS_INVALID, engine.InvalidParams.With(err)
+func (api *ConsensusAPI) ForkchoiceUpdatedV2(update engine.ForkchoiceStateV1, params *engine.PayloadAttributes) (engine.ForkChoiceResponse, error) {
+	if params != nil {
+		if params.Withdrawals == nil {
+			return engine.STATUS_INVALID, engine.InvalidParams.With(errors.New("missing withdrawals"))
+		}
+		if params.BeaconRoot != nil {
+			return engine.STATUS_INVALID, engine.InvalidParams.With(errors.New("unexpected beacon root"))
+		}
+		c := api.eth.BlockChain().Config()
+		if !active(c.IsShanghai, c.IsCancun, c.LondonBlock, params.Timestamp) {
```

Comment:
It feel like there are two problems here:

1) Is don't have a way of referring to a fork beyond `IsX`. So there is no way to verify mutual exclusion of two forks without the other. If we had say `forks.Cancun` as a identifier, maybe the function could look much simpler.
2) Time-based forks also require the block number, which further makes this line look unintuitive. I used `c.LondonBlock` since other areas of code are focused on not accepting engine api requests if we're pre-ttd. We could change it to the current block number, but probably does little to improve clarity.

Let me mull over it some more.